src/umleditor/mvc_model/entity.py

```diff
@@ -48,6 +48,22 @@ def add_field(self, field_name:str):
             raise CustomExceptions.FieldExistsError(field_name)
         else:
             self._fields.append(field_name)
+    
+    def edit_field(self, old_field: str, new_field: str):
```

Comment:
why does the method rename_field not work here?

Reply:
^if the method rename_field works here, delete this method.

Reply:
The edit_field allows for passing an old and new name without errors, while the rename does not, this functionality is necessary with the current GUI Design.

Reply:
To be clear, we are both talking about the rename_field on line 71 of entity.py, correct? If so, the functions are literally identical. Error checking doesn't matter if the error cannot be raised, so there's no reason to make a separate method to exclude it.

Reply:
Yes, I am referring to the  def rename_field() method. The key difference between edit and rename is that within rename it is considered illegal to rename a field given the same name, hence it raises an error. My edit_field function does not consider this action illegal and will not throw an exception. I.E if I want to edit something and I don't wind up changing the values, that's okay within edit_field. It is not okay to perform this edit within rename_field() and will stop the flow of the GUI. Although the difference in code is minor, the methods are not interchangeable within the GUI.

Reply:
you can add "if old_name == new_name: return" to rename_field. Setters conventionally do nothing if old and new are the same. It won't break the CLI.

Reply:
rename_field has been modified and edit_field and related files have been deleted